app/policies/application_policy.rb

```diff
@@ -0,0 +1,95 @@
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  @scopes_by_action = {}
```

Comment:
can't you just use the `scopes_by_action` class method to access this variable with a default instead of evaluating on load?